### PR TITLE
Add Laird Connectivity BL5340PA DVK board

### PR DESCRIPTION
**Pull request by ahedin25**

### CODE REVIEW

File: boards/arm/bl5340_dvk/doc/index_bl5340pa.rst [hunk 1]
```diff
@@ -0,0 +1,454 @@
+.. _bl5340pa_dvk:
+
+Laird Connectivity BL5340PA DVK
+###############################
+
+Overview
+********
+The BL5340PA Development Kit provides support for the Laird
+Connectivity BL5340PA module which is powered by a dual-core Nordic
+Semiconductor nRF5340 ARM Cortex-M33F CPU and a Nordic Semiconductor
```
Comment (rerickson1) [inline boards/arm/bl5340_dvk/doc/index_bl5340pa.rst:10]: Isn't just `M33` not `M33F`

File: boards/arm/bl5340_dvk/bl5340pa_dvk_cpunet.dts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+	nrf_radio_fem: nrf21540_fem {
+		compatible = "nordic,nrf21540-fem";
+		rx-en-gpios = <&gpio0 31 GPIO_ACTIVE_HIGH>;
+		tx-en-gpios = <&gpio0 30 GPIO_ACTIVE_HIGH>;
+		pdn-gpios = <&gpio0 29 GPIO_ACTIVE_HIGH>;
+		supply-voltage-mv = <3300>;
+	};
+
+	lcz_fem: nrf21540_lcz_fem  {
+		compatible = "lairdconnect,nrf21540-lcz-fem";
```
Comment (rerickson1) [inline boards/arm/bl5340_dvk/bl5340pa_dvk_cpunet.dts:42]: Will there be a commit to add the DTS bindings and driver for this?

**Review by rerickson1 — CHANGES_REQUESTED** [review]

File: boards/arm/bl5340_dvk/doc/index_bl5340pa.rst (hunk 1 above)
Comment (ahedin25) [reply boards/arm/bl5340_dvk/doc/index_bl5340pa.rst:10]: I believe this was copied from Nordic thingy53.  Do you think source is incorrect?

File: boards/arm/bl5340_dvk/bl5340pa_dvk_cpunet.dts (hunk 2 above)
Comment (ahedin25) [reply boards/arm/bl5340_dvk/bl5340pa_dvk_cpunet.dts:42]: Yes, but it is going to be part of NCS.

File: boards/arm/bl5340_dvk/doc/index_bl5340pa.rst (hunk 1 above)
Comment (greg-leach) [reply boards/arm/bl5340_dvk/doc/index_bl5340pa.rst:10]: Datasheet reads as follows.

nRF5340 SoC Product Brief Version 2.2
For more information please visit: nordicsemi.com/nRF5340
Key features
 High-performance application processor
 128/64 MHz Arm Cortex-M33 with
FPU and DSP instructions
 1 MB Flash + 512 KB RAM
 8 KB 2-way set associative cache

 Fully-programmable network processor
 64 MHz Arm Cortex-M33 with
2 KB instruction cache
 256 KB Flash + 64 KB RAM

Reading this as the application core being 33F and the network core plain 33.

**Review by greg-leach — DISMISSED** [review]

File: boards/arm/bl5340_dvk/doc/index_bl5340pa.rst (hunk 1 above)
Comment (rerickson1) [reply boards/arm/bl5340_dvk/doc/index_bl5340pa.rst:10]: Hmm yes, seems ok.

**Pull request closed without merging**